src/dns.c

```diff
@@ -3351,7 +3196,7 @@ hsk_dns_rrs_clean(hsk_dns_rrs_t *rrs, uint16_t type) {
  */
 
 bool
-hsk_dns_is_subdomain(const char *parent, const char *child) {
+hsk_dns_is_subdomain(const uint8_t *parent, const uint8_t *child) {
```

Comment:
While playing with this method I wrote the tests for this and it fails when `child == parent`.

Test case can be included in `test/hnsd-test.c`. Third test will fail both on master and this branch.
So this bug is not related to the PR, but in general to this method.
```c
void
test_hsk_dns_is_subdomain() {
  printf("  test_hsk_dns_is_subdomain\n");

  typedef struct {
    uint8_t parent[255];
    uint8_t child[255];
    bool result;
  } hsk_dns_is_subdomain_t;

  const hsk_dns_is_subdomain_t tests[] = {
    {
      .parent = "\x09""handshake""\x03""org""\x00",
      .child = "\x04""hnsd""\x09""handshake""\x03""org""\x00",
      .result = true
    },
    {
      .parent = "\x04""hnsd""\x09""handshake""\x03""org""\x00",
      .child = "\x09""handshake""\x03""org""\x00",
      .result = false
    },
    {
      .parent = "\x09""handshake""\x03""org""\x00",
      .child = "\x09""handshake""\x03""org""\x00",
      .result = true
    },
    {
      .parent = "\x09""handshake""\x03""org""\x00",
      .child = "\x04""test""\x03""org""\x00",
      .result = false
    }
  };

  for (int i = 0; i < ARRAY_SIZE(tests); i++) {
    const hsk_dns_is_subdomain_t test = tests[i];
    bool result = hsk_dns_is_subdomain(test.parent, test.child);
    assert(result == test.result);
  }
}
```
Javascript version will return expected results.
https://github.com/chjj/bns/blob/1bc5d056f4372e62cefd43112ca8b1ebc3372f86/lib/util.js#L317-L319

Reply:
Confirmed that equal name should count here looking at knot resolver tests (and their `knot_dname_in_bailiwick()` function)

fixed and added test in fe6e1d5ea0439edc7fcd2d6e9668e3dd1c80ea9c